src/cpp/include/UTIL/LCRelationNavigator.h

```diff
@@ -69,24 +70,61 @@ namespace UTIL {
     const EVENT::FloatVec & getRelatedFromWeights(EVENT::LCObject * to) const ;
 
     /** Object with a highest weight that the given from-object is related to.
-     *  LCObject is of type getToType().
+     *  LCObject is of type getToType(). Different comparator function can be specified
+     *  in a second argument analogous to e.g. std::min() overload.
      */
-    const EVENT::LCObject* getRelatedToMaxWeightObject(EVENT::LCObject* from, const std::string& weightType) const ;
+    template <typename CompareF = std::less<float> >
+    const EVENT::LCObject* getRelatedToMaxWeightObject(EVENT::LCObject* from, CompareF&& compare = CompareF() ) const{
+        const auto& objects = getRelatedToObjects(from);
+        if ( objects.empty() ) return nullptr;
+
+        const auto& weights = getRelatedToWeights(from);
+        const auto maxWeightIt = std::max_element(weights.begin(), weights.end(), compare);
+        int i = std::distance(weights.begin(), maxWeightIt);
+        return objects[i];
+    }
 
     /** From-object related to the given object with a highest weight (the inverse relationship).
-     *  LCObject is of type getFromType().
+     *  LCObject is of type getFromType(). Different comparator function can be specified
+     *  in a second argument analogous to e.g. std::min() overload.
      */
-    const EVENT::LCObject* getRelatedFromMaxWeightObject(EVENT::LCObject* to, const std::string& weightType) const ;
+    template <typename CompareF = std::less<float> >
+    const EVENT::LCObject* getRelatedFromMaxWeightObject(EVENT::LCObject* to, CompareF&& compare = CompareF() ) const{
+        const auto& objects = getRelatedToObjects(to);
```

Comment:
```suggestion
        const auto& objects = getRelatedFromObjects(to);
```
I am not entirely sure, because also the old version used `getRelatedToObjects`

Reply:
omg, nice catch... It was definitely a bug :confused: